app/src/main/java/org/schabi/newpipe/settings/ContentSettingsManager.kt

```diff
@@ -48,6 +49,33 @@ class ContentSettingsManager(private val fileLocator: NewPipeFileLocator) {
         return fileLocator.dbDir.exists() || fileLocator.dbDir.mkdir()
     }
 
+    fun isValidBackupFile(filePath: String): Boolean {
+        try {
+            val zipFile = ZipFile(filePath)
+            if (zipFile.size() != 2)
+                return false
+
+            val entries = zipFile.entries()
+            var containsDbFile = false
+            var containsSettingsFile = false
+
+            while (entries.hasMoreElements()) {
+                val entry = entries.nextElement()
+                if (entry.name == "newpipe.db")
+                    containsDbFile = true
+                else if (entry.name == "newpipe.settings")
+                    containsSettingsFile = true
+            }
+
+            return containsDbFile && containsSettingsFile
```

Comment:
`return zipfile.entries().toList().map { it.name }.containsAll(listOf("newpipe.db", "newpipe.settings"))`

should do the same and makes use of kotlin features to write more concise code.

Reply:
Okay, I look into it (many list-comprehensions xD).